chat/models.tsp

```diff
@@ -204,149 +183,97 @@ model CreateChatCompletionResponse {
-  prompt_tokens: safeint;
-
-  /** Number of tokens in the generated completion */
-  completion_tokens: safeint;
-
-  /** Total number of tokens used in the request (prompt + completion). */
-  total_tokens: safeint;
-}
-
 model ChatCompletionTool {
-  /** The type of the tool. Currently, only `function` is supported. */
+  @doc("""
+  The type of the tool. Currently, only `function` is supported.
+  """)
   type: "function";
 
   function: FunctionObject;
 }
 
-/**
- * Controls which (if any) function is called by the model. `none` means the model will not call a
- * function and instead generates a message. `auto` means the model can pick between generating a
- * message or calling a function. Specifying a particular function via
- * `{"type": "function", "function": {"name": "my_function"}}` forces the model to call that
- * function.
- * 
- * `none` is the default when no functions are present. `auto` is the default if functions are
- * present.
- */
-@oneOf
-@extension("x-oaiExpandable", true)
-union ChatCompletionToolChoiceOption {
-  "none",
-  "auto",
-  ChatCompletionNamedToolChoice,
-}
-
 /** Specifies a tool the model should use. Use to force the model to call a specific function. */
 model ChatCompletionNamedToolChoice {
-  /** The type of the tool. Currently, only `function` is supported. */
+  @doc("""
+  The type of the tool. Currently, only `function` is supported.
+  """)
   type: "function";
 
   function: {
     /** The name of the function to call. */
     name: string;
-  }
+  };
 }
 
-@oneOf
-union ChatCompletionRequestUserMessageContent {
-  /** The text contents of the message. */
-  string,
+@doc("""
+Controls which (if any) function is called by the model.
+`none` means the model will not call a function and instead generates a message.
+`auto` means the model can pick between generating a message or calling a function.
+Specifying a particular function via `{"type": "function", "function": {"name": "my_function"}}` forces the model to call that function.
 
-  /**
-   * An array of content parts with a defined type, each can be of type `text` or `image_url` when
-   * passing in images. You can pass multiple images by adding multiple `image_url` content parts.
-   * Image input is only supported when using the `gpt-4-visual-preview` model.
-   */
-  ChatCompletionRequestMessageContentParts,
-};
+`none` is the default when no functions are present. `auto` is the default if functions are present.
+""")
+@extension("x-oaiExpandable", true)
+union ChatCompletionToolChoiceOption {
+"none" | "auto",
+ChatCompletionNamedToolChoice
+}
 
-@minItems(1)
-model ChatCompletionRequestMessageContentParts is ChatCompletionRequestMessageContentPart[];
 
-@oneOf
 @extension("x-oaiExpandable", true)
 union ChatCompletionRequestMessageContentPart {
-  ChatCompletionRequestMessageContentPartText,
-  ChatCompletionRequestMessageContentPartImage,
+ChatCompletionRequestMessageContentPartText,
+ChatCompletionRequestMessageContentPartImage
 }
```

Comment:
nit: Unions are missing indentation.

Reply:
I must've missed running `tsp format` before the last push! Should be addressed, and several other things are probably more readable, too. Sorry about that!